packages/xo-server/src/xo-mixins/xen-servers.js

```diff
@@ -396,9 +439,10 @@ export default class {
     }
 
     delete this._xapis[id]
-    delete this._serverIdsByPool[xapi.pool.$id]
-
-    xapi.xo.uninstall()
+    if (xapi.status === 'connected') {
```

Comment:
Don't mix up `xen-api` status and our own (`xo-server's status).

Don't rely on `xapi.status` to check if the server is fully connected.